README.md

```diff
@@ -1,82 +1,65 @@
 
-def geometric_mean(x: Var, y: Var) -> Var:
-    return op.sqrt(
-      op.mul(x, y)
-    )
+```bash
+conda install spox
 ```
 
-Since ONNX is tensor-oriented, this code assumes that `x` and `y` are floating point tensors with matching (broadcastable) shapes, and the geometric mean is computed elementwise. You may learn more about what types and shapes are acceptable for given operators by checking their docstring.
+## Quick start
 
-Performing operations on Variables creates further Variables, all of which keep track of what computation has been performed so far. If an operation is determined to be illegal (for example due to mismatching types/shapes), an exception will be immediately raised by Spox.
+Spox users interact most commonly with `Var` objects - **variables** which will have values assigned to them at runtime.
+The initial `Var` objects which represent the *argument*s of a model (the model inputs in ONNX nomenclature) are created with an explicit type and shape using the `spox.argument` function.
+All further `Var` objects are created by calling functions which take existing `Var` objects as inputs and produce new `Var` objects as outputs.
+Spox provides such functions for all operators in the standard grouped by domain and version inside the `spox.opset` module.
 
-### Building
+The final `onnx.ModelProto` object is build by specifying the explicit input and output `Var`s of the model to in the `spox.build` function.
 
-To introduce _argument_ variables (which are placeholders for runtime values - graph/model inputs in ONNX nomenclature), you may use the `argument(typ: Type) -> Var` function with the required type, like `Tensor`.
-
-When you're done you may build an `onnx.ModelProto` - the model protobuf, which is the ONNX program representation. You may do this with the public `build(inputs, outputs) -> onnx.ModelProto` function:
+Below is an example for defining an ONNX graph which computes the [geometric mean](https://en.wikipedia.org/wiki/Geometric_mean) of two inputs.
+Please consult the [documentation](https://spox.readthedocs.io/en/latest) for more details and further tutorials on how to use Spox.
 
 ```python
+import onnx
+
 from spox import argument, build, Tensor, Var
+# import operators from the ai.onnx domain at version 17
+from spox.opset.ai.onnx import v17 as op
 
-def geometric_mean(x: Var, y: Var) -> Var: ...
+def geometric_mean(x: Var, y: Var) -> Var:
+    # use the standard Sqrt and Mul
+    return op.sqrt(op.mul(x, y))
 
-a = argument(Tensor(float, (1, 'N')))
-b = argument(Tensor(float, ('M', 1)))
+# Create typed model inputs. Each tensor is of rank 1
+# and has the runtime-defined length "N".
+a = argument(Tensor(float, ('N',)))
+b = argument(Tensor(float, ('N',)))
 
+# Perform operations on `Var`s
 c = geometric_mean(a, b)
 
-model = build({'a': a, 'b': b}, {'c': c})
+# Build an `onnx.ModelProto` for the given inputs and outputs.
+model: onnx.ModelProto = build(inputs={'a': a, 'b': b}, outputs={'c': c})
```

Comment:
I don't see why writing out the argument names is necessary? It already goes left-to-right, so inputs-then-outputs seems intuitive enough.
```suggestion
model: onnx.ModelProto = build({'a': a, 'b': b}, {'c': c})
```

Reply:
I think being a bit more explicit in the example does not hurt.

Reply:
Fair enough